applications/nrf_desktop/doc/hid_forward.rst

```diff
@@ -32,38 +32,66 @@ Complete the following steps to configure the module:
+This means that the |hid_forward| can concurrently send reports to every available HID-class USB device, if a report from the linked peripheral is available.
 
 Forwarding HID input reports
 ============================
 
-After :ref:`nrf_desktop_ble_discovery` successfully discovers a connected peripheral, the HID forward module automatically subscribes for every HID input report provided by the peripheral.
+After :ref:`nrf_desktop_ble_discovery` successfully discovers a connected peripheral, the |hid_forward| automatically subscribes for every HID input report provided by the peripheral.
 
 The :cpp:func:`hidc_read` callback is called when HID input report is received from the connected peripheral.
 The received HID input report data is converted to ``hid_report_event``.
 
-``hid_report_event`` is submitted and then the :ref:`nrf_desktop_usb_state` forwards it to the host.
-When a HID report is sent to the host, the ``hid_forward`` module receives a ``hid_report_sent_event``.
+``hid_report_event`` is submitted and then the HID-class USB device configured by :ref:`nrf_desktop_usb_state` forwards it to the host.
+When a HID report is sent to the host by the HID-class USB device, the |hid_forward| receives a ``hid_report_sent_event`` with the identifier of this device.
 
 Enqueuing incoming HID input reports
 ====================================
 
-The device forwards only one HID input report to the host at a time.
-Another HID input report may be received from a peripheral connected over Bluetooth before the previous one was sent to the host.
+The |hid_forward| forwards only one HID input report to the HID-class USB device at a time.
+Another HID input report may be received from a peripheral connected over Bluetooth before the previous one was sent.
 In that case, ``hid_report_event`` is enqueued and submitted later.
-Up to ``CONFIG_DESKTOP_HID_FORWARD_MAX_ENQUEUED_REPORTS`` reports can be enqueued at a time.
-In case there is no space to enqueue a new event, the module drops the oldest enqueued event.
+Up to ``CONFIG_DESKTOP_HID_FORWARD_MAX_ENQUEUED_REPORTS`` reports can be enqueued at a time for each report type and for each connected peripheral.
+If there is not enough space to enqueue a new event, the module drops the oldest enqueued event that was received from this peripheral (of the same type).
 
-Upon receiving the ``hid_report_sent_event``, ``hid_forward`` submits the first enqueued ``hid_report_event``.
+Upon receiving the ``hid_report_sent_event``, the |hid_forward| submits the ``hid_report_event`` enqueued for the peripheral that is associated with the HID-class USB device.
+The enqueued report to be sent is chosen by the |hid_forward| in the round-robin fashion.
+The report of the next type will be sent if available.
+If not available, the next report type will be checked until no report is found or there are no enqueued reports.
```

Comment:
@pdunaj , isn't `until no report is found or there are no enqueued reports` basically the same?